zingolib/src/lightclient.rs

```diff
@@ -1522,11 +1562,11 @@ impl LightClient {
                 .await;
 
         // Do Trial decryptions of all the outputs, and pass on the successful ones to the update_notes processor
-        let trial_decryptions_processor = TrialDecryptions::new(
+        let trial_decryptions_processor = Arc::new(TrialDecryptions::new(
```

Comment:
Add a comment explaining why the tdp now must be in an `Arc`

Reply:
Turns out there's no longer a good reason!